src/metrics.jl

```diff
@@ -348,31 +397,31 @@ end
 end
 function eval_end(::CosineDist, s)
     ab, a2, b2 = s
-    max(1 - ab / (sqrt(a2) * sqrt(b2)), zero(eltype(ab)))
+    max(1 - ab / (sqrt(a2) * sqrt(b2)), 0)
 end
 cosine_dist(a::AbstractArray, b::AbstractArray) = CosineDist()(a, b)
+cosine_dist(a::Number, b::Number) = CosineDist()(a, b)
 
-# Correlation Dist
+# CorrDist
 _centralize(x::AbstractArray) = x .- mean(x)
 (dist::CorrDist)(a::AbstractArray, b::AbstractArray) = CosineDist()(_centralize(a), _centralize(b))
+(dist::CorrDist)(a::Number, b::Number) = CosineDist()(zero(mean(a)), zero(mean(b)))
 corr_dist(a::AbstractArray, b::AbstractArray) = CorrDist()(a, b)
-result_type(::CorrDist, Ta::Type, Tb::Type) = result_type(CosineDist(), Ta, Tb)
+corr_dist(a::Number, b::Number) = CorrDist()(a, b)
+# result_type(::CorrDist, Ta::Type, Tb::Type) = result_type(CosineDist(), Ta, Tb)
```

Comment:
?

Reply:
Sorry, I forgot to delete it. This is simply unnecessary, because the generic code already does the job.